packages/@aws-cdk/aws-stepfunctions-tasks/lib/codebuild/start-build.ts

```diff
@@ -109,4 +183,17 @@ export class CodeBuildStartBuild extends sfn.TaskStateBase {
       Value: environmentVariables[name].value,
     }));
   }
+
+  private validateOverridingParameters(props: CodeBuildStartBuildProps) {
+    if (props.overrides?.secondaryArtifacts && props.overrides!.secondaryArtifacts!.length > 12) {
+      throw new Error('The maximum array length of override property "secondaryArtifacts" is 12.');
```

Comment:
is this being tested somewhere? - I think we can also improve on the error message.
how about indicating that the limit was reached and what we received

```suggestion
      throw new Error(`The maximum overrides that can be specified for 'secondaryArtifacts' is 12. Received: ${props.overrides!.secondaryArtifacts!.length}`);
```